web-server/src/grading-queue/get.ts

```diff
@@ -1,29 +1,72 @@
-import { client } from "../index";
+import { GradingJob } from "./types";
+import { getSubmitterInfo } from "../utils/helpers";
+import { redisGet, redisZRangeWithScores } from "../utils/redis";
 
-const getGradingJobsFromKeys = async (grading_info_keys: string[]) => {
-  const grading_jobs = grading_info_keys.map((key) => client.get(key));
-  const results = await Promise.all(grading_jobs);
-  return results;
-};
+const KEY_DELIM = ".";
 
-const getGradingQueue = async () => {
-  const grading_info_keys: string[] = await client.keys("QueuedGradingInfo.*");
-  // TODO: Make type model for grading job info
-  //   const grading_jobs = await getGradingJobsFromKeys(grading_info_keys);
-  let grading_jobs: Object[] = [];
-  for (let i = 0; i < grading_info_keys.length; i++) {
-    const key = grading_info_keys[i];
-    const redis_grading_job = await client.get(key);
-    if (!redis_grading_job) {
-      // Error retrieving grading job
-      // TODO: If getting a key fails for some reason -- returns null
-      // do we want the rest of the list?  Or just say it failed?
-      return;
-    }
-    const grading_job = JSON.parse(redis_grading_job);
-    grading_jobs.push(grading_job);
+// Get grading jobs with their corresponding nonces
+const getCollatedGradingJobs = async (): Promise<
+  [GradingJob[] | null, Error | null]
+> => {
+  const [reservations, reservationsErr] = await redisZRangeWithScores(
+    "Reservations",
+    0,
+    -1,
+  );
+  if (reservationsErr) {
+    return [null, reservationsErr];
```

Comment:
Just to confirm, I imagine we'll be phasing out this data/error return structure in a future PR?

Reply:
Yeah I think that would be good.  I haven't done it yet since the PRs I have been working on are primarily front end work.